### ISSUE TITLE
Alphafold3

### ISSUE THREAD

**Issue by WilleBell** [vscentrum/vsc-software-stack#454]
* link to support ticket: [VUB request](https://vub.service-now.com/now/nav/ui/classic/params/target/incident.do%3Fsys_id%3D41bbe612c3395e10443d7ac4e401318e%26sysparm_record_target%3Dincident%26sysparm_record_row%3D1%26sysparm_record_rows%3D6%26sysparm_record_list%3Dactive%253Dtrue%255Eassigned_to%253D8b8ed3c997b2e55075ecb7b3f153af44%255Estate%2521%253D6%255EORDERBYDESCsys_updated_on), [Other request (KUL?)](https://otrsdict.ugent.be/otrs/index.pl?Action=AgentTicketZoom;TicketID=175941)
* website: https://github.com/google-deepmind/alphafold3
* installation docs: 
* toolchain: 
* easyblock to use: 
* required dependencies:
  * absl-py
  * chex
  * dm-haiku==0.0.13
  * dm-tree
  * jax==0.4.34
  * jax[cuda12]==0.4.34
  * jax-triton==0.2.0
  * jaxtyping
  * numpy
  * rdkit==2024.3.5
  * triton==3.1.0
  * tqdm
  * zstandard
* optional dependencies:
  * CUDA: 12.6
* notes:
  * We might want to land this in `2024a` ?
  * They provide a dockerfile (see below)
* effort: *(TBD)*
* other install methods
  * conda: no
  * container image: yes (https://github.com/google-deepmind/alphafold3/tree/v3.0.0/docker)
  * pre-built binaries (RHEL8 Linux x86_64): no
  * easyconfig outside EasyBuild: no

**Comment by lexming** [comment]
Updated the dependencies based on what is in its [pyproject.toml](https://github.com/google-deepmind/alphafold3/blob/main/pyproject.toml). The file [requirements.txt](https://github.com/google-deepmind/alphafold3/blob/main/requirements.txt) has a more exhaustive list with all the versions pinned to what they probably use internally for testing.

**Comment by boegel** [comment]
We'll probably get attacked from multiple angles again for this one, so perhaps it's worth setting up a "task force" and having a dedicated short meeting to discuss how to quickly get this supported into EasyBuild?

Other people outside of VSC (like @ThomasHoffmann77) may be interested in helping out here...

**Comment by ThomasHoffmann77** [comment]
@jpecar

**Comment by jpecar** [comment]
Yes, 2024a seems like a proper toolchain for this. However it will need some work, first major one will be SciPy-bundle. See https://github.com/easybuilders/easybuild-easyblocks/issues/3508
Jax and Triton are also potential pain points, others look doable at this stage.

**Comment by lexming** [comment]
I would focus first on Jax/Triton and try to use the already available [SciPy-bundle-2024.05-gfbf-2024a.eb](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/SciPy-bundle/SciPy-bundle-2024.05-gfbf-2024a.eb) with `numpy==1.26.4`. Neither Jax or AlphaFold3 have a requirement on `numpy>=2.0` in their `pyproject.toml`, so they might work with 1.26.4.

**Comment by jpecar** [comment]
What about requirements.txt and dev-requirements.txt ? Are these still relevant?

**Comment by lexming** [comment]
Those are used to [populate the container](https://github.com/google-deepmind/alphafold3/blob/2ffe43f574436e6473bf3ab1c0c61c040edc59fd/docker/Dockerfile#L44) and I guess it also reflects what is used internally for testing. So I agree that it would be nice if we can stick to those same versions, but IMO not mandatory if the code works with what we already have.

**Comment by WilleBell** [comment]
There's a [WIP PR](https://github.com/easybuilders/easybuild-easyconfigs/pull/21864) for AlphaFold3 in `2024a`.

**Comment by boegel** [comment]
PR for AlphaFold v3.0.1 is merged:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/24177

I started looking into updating that to AlphaFold v3.0.4 (latest current), but it seems like that'll require a toolchain jump too:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/26779

@pavelToman Can you check how much effort it would be to port https://github.com/easybuilders/easybuild-easyconfigs/pull/26779 over to either `2025a` or `2025b` (whatever is easiest, probably `2025a` since `jax-0.7.0-gfbf-2025a-CUDA-12.8.0.eb` already exists)
